src/dstack/_internal/core/backends/digitalocean_base/compute.py

```diff
@@ -88,15 +88,18 @@ def create_instance(
             name=f"dstack-{instance_config.project_name}",
             public_key=project_ssh_key.public,
         )
-
         size_slug = instance_offer.instance.name
 
         if not instance_offer.instance.resources.gpus:
             backend_specific_commands = SETUP_COMMANDS + DOCKER_INSTALL_COMMANDS
         else:
             backend_specific_commands = SETUP_COMMANDS
 
-        project_id = self.api_client.get_project_id(self.config.project_name)
+        project_id = (
+            self.api_client.get_project_id(self.config.project_name)
+            if self.config.project_name
+            else None
+        )
```

Comment:
It is still worth checking that `get_project_id` didn't return `None`. This would mean that `project_name` no longer exists, in which case provisioning should fail, not be redirected to the default project.

```suggestion
        project_id = None
        if self.config.project_name:
            project_id = self.api_client.get_project_id(self.config.project_name)
            if project_id is None:
                raise BackendError(f"Project {self.config.project_name} does not exist")
```